### DISCUSSION TITLE
Fastapi Call websocket

### DISCUSSION THREAD

**Discussion by Frenz86** [zauberzeug/nicegui#3021] [Q&A] [answered]
Hi, i modify your template in order to use nicegui as fastapi backend server with the frontend. It works very well,
with also multipage, but when i modify input, the async call get only default values...
Where's the error? thank you so much for this amzing framework.

https://github.com/Frenz86/fastapinicegui

BR
Frenz

**Comment by rodja** [comment]
Sorry. Could you break it down and create a minimal example which shows your problem?

**Reply by Frenz86** [reply]
Hi, basically it didn't update the value on ui.markdown, but only on ui.notify
![problem](https://github.com/zauberzeug/nicegui/assets/25400238/0ef01cc5-edf4-47e9-9c68-8d6396498bdc)

**Comment by falkoschindler** [comment, marked as answer]
@Frenz86 I roughly boiled it down to the following code, even though the FastAPI part isn't needed to reproduce the problem:
```py
import uvicorn
from fastapi import FastAPI
from nicegui import ui

app = FastAPI()

async def compute_and_update_result(num1, num2):
    somma = num1 + num2
    ui.notify(f'risultato : {somma}')
    return somma

@ui.page('/')
async def home():
    num1 = ui.number(value=1)
    num2 = ui.number(value=2)
    ui.button('risultato', on_click=lambda: compute_and_update_result(num1.value, num2.value))
    result = await compute_and_update_result(num1.value, num2.value)
    ui.markdown(f"#### result = {result}")  # didn't update

ui.run_with(app, mount_path='/')

if __name__ == "__main__":
    uvicorn.run("main:app", host="127.0.0.1", port=8000, reload=True)
```

You create a `ui.markdown` element when generating the page, but never update its content. The line `result = await compute_and_update_result(num1.value, num2.value)` only runs once, not every time the button is clicked.

Something like this might work:
```py
@ui.page('/')
async def home():
    num1 = ui.number(value=1)
    num2 = ui.number(value=2)

    async def handle_click():
        result = await compute_and_update_result(num1.value, num2.value)
        markdown.content = f"#### result = {result}"

    ui.button('risultato', on_click=handle_click)
    markdown = ui.markdown()
```

**Reply by Frenz86** [reply]
Guys you made my day!! Now i can use fastapi for ML inference and nicegui frontend to make the UI interface, thank you so much!
